exercises/practice/prime number/prime_example.py

```diff
@@ -0,0 +1,18 @@
+from z3 import *
+
+def prime_number(x):
+
+    a, b = Ints('a b')
+    s1 = Solver()
+    s1.add(And(x > 1, (x % 1) == 0))
+
+    if(s1.check() == unsat):
+        return "not prime"
+
+    s2 = Solver()
```

Comment:
Instead of creating two separate solvers, you can use the push() and pop() Solver methods.  They behavior similarly to pushing and popping data to/from a stack.  You can look how I used them here: [https://github.com/exercism/z3/blob/main/exercises/practice/equilateral_triangle/equilateral-triangle-example.py#L40](url).  You may be able to combine them into one solver anways though.  May be worth testing that out.

Reply:
Much cleaner method. Did not improve run time but cleaned up code. Thanks!